src/components/LetterDetail.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { CloudIcon, ColorSirenIcon, SnowIcon, ThermostatIcon, WarmIcon } from '@/assets/icons';
+
+import ReportModal from './ReportModal';
+
+export default function LetterDetail({ title, text }: { title: string; text: string }) {
+  const FONT = '';
+  const THEME = 'celebrate';
+  const DEGREES = [
+    { icon: WarmIcon, title: '따뜻해요' },
+    { icon: CloudIcon, title: '그럭저럭' },
+    { icon: SnowIcon, title: '앗! 차가워' },
+  ];
+  const [degreeModalOpen, setDegreeModalOpen] = useState<boolean>(false);
+  const [reportModalOpen, setReportModalOpen] = useState<boolean>(false);
+
+  const degreeButtonRef = useRef<HTMLButtonElement>(null);
+  const handleOutsideClick = (event: MouseEvent) => {
+    const target = event.target as Node;
+    if (!target || degreeButtonRef.current?.contains(target)) {
+      return;
+    }
+    setDegreeModalOpen(false);
+  };
+  useEffect(() => {
+    document.body.addEventListener('click', handleOutsideClick);
+
+    return () => {
+      document.body.removeEventListener('click', handleOutsideClick);
+    };
+  }, []);
+  return (
+    <>
+      {reportModalOpen && <ReportModal onClose={() => setReportModalOpen(false)} />}
+      <div className={twMerge(`flex grow flex-col gap-3 px-5 pb-7.5`, THEME)}>
+        <div className="absolute top-5 right-5 flex gap-3">
+          <button
+            ref={degreeButtonRef}
+            className="flex items-center justify-center gap-1"
+            onClick={() => {
+              setDegreeModalOpen((cur) => !cur);
+            }}
+          >
+            <img src={ThermostatIcon} alt="편지 온도 아이콘" />
+            <span className="caption-b text-primary-1">편지 온도</span>
+          </button>
+          <button
+            onClick={() => {
+              setReportModalOpen(true);
+            }}
+          >
+            <img src={ColorSirenIcon} alt="신고 아이콘" />
```

Comment:
이거 컴포넌트로 사용하도록 만들어뒀어요! 이렇게 해도 문제는 없겠지만
`<ColorSirenIcon />`으로 사용 가능합니다!!

그리고 제가 `<SirenOutlinedIcon />`을 하나 만들어두었어요. 이렇게 하면 className으로 아이콘 색상 지정도 가능합니다!
예를 들면 이렇게요!
```suggestion
            <SirenOutlinedIcon className="text-primary-1" />
```

이렇게 했을 때의 장점은!! svg 에셋이 줄어든다는 것이지용~~
컬러마다 svg 추출없이 하나의 svg만 뽑아서 색상 변경만 하면 되니까요!

Reply:
앗차차! 아이콘 크기가 다양할 수 있어서 width랑 height는 반드시 포함시키도록 해놔서..
제가 제안드린 방식대로 하신다면 className에 width랑 height 반드시 넣어주셔야 합니다...ㅎㅎㅎ

Reply:
오오오 이게 svgr이군요 ㄷㄷㄷㄷ 잘 적용해서 사용해보겠습니당 감사합니다!!!👍👍👍👍👍